### PR TITLE
Kendall - discount codes routes and services

### PR DESCRIPTION
**Pull request by KENDARRR**
## Brief Description

Discount code API added. GET, DELETE, and POST are routes are active once the new db gets merged. Delete simply changes the number us uses left to 0 so no one else can the coupon code. The others work as expected.

## References Issue

Issue 103

## Type of change

- [ ] Bug fix
- [!!!] Feature
- [ ] Major change (potentially breaking, list anything that is broken below)
- [ ] Documentation

### Breaking Changes

Incompatible with current db. Will work once the new db is merged
## Testing done

Describe tests that were used, along with instructions for testing.

- [ !] Postman GET route
- [ !] Postman DELETE route
- [ !] Postman POST route

- [ !] Docker was used.
- [ ] Docker was not used and I have listed my configuration below.

**System Specifications**:
* Operating System: Win 11 Home
* Node Version: 16.16.0
* NPM Version: 8.18.0

## Checklist:

- [! ] My code follows the styling guidelines.
- [ !] I have added unit tests and verified that they pass.
- [ ] I have used the linter and fixed any linting issues.
- [! ] I have commented the code.
- [ !] I have adjusted the documentation to match the changed code.
- [! ] Prior to submitting the PR, I made sure the latest changes were merged with my branch.
- [ !] Existing tests pass locally with changes.

### Explanation for unchecked

I'm checking with Ben about Linting right now

### CODE REVIEW

File: server/src/api/discounts/discounts.router.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+});
+
+
+/**
+ * route: POST /
+ *
+ * @type {?}
+ */
+
+discountsRouter.post('/', async(
```
Comment (ben16williams) [inline server/src/api/discounts/discounts.router.ts:45]: This should probably be a private route

File: server/src/api/discounts/discounts.router.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    }
+});
+
+/**
+ * route: DELETE /
+ *
+ * @type {?}
+ */
+
+discountsRouter.delete('/:id', async(
```
Comment (ben16williams) [inline server/src/api/discounts/discounts.router.ts:64]: This should also probably be private

File: server/src/api/discounts/discounts.router.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+/**
+ * create Router object
+ *
+ * @type {?}
+ */
+
+export const discountsRouter = Router();
+//discountsRouter.use(checkJwt);
+//discountsRouter.use(checkScopes);
```
Comment (ben16williams) [inline server/src/api/discounts/discounts.router.ts:17]: Do we need these as comments here?

**Review by ben16williams — CHANGES_REQUESTED** [review]
Nice job with documentation

**Comment by ben16williams** [comment]
Can you also make the title of the PR something that shortly describes the work?

File: server/src/api/discounts/discounts.router.ts (hunk 3 above)
Comment (KENDARRR) [reply server/src/api/discounts/discounts.router.ts]: All should be fixed now

**Review by ben16williams — APPROVED** [review]

**Pull request merged**